dialoguekit/utils/dialogue_evaluation.py

```diff
@@ -141,10 +153,7 @@ def reward(self) -> Dict[str, List[Dict[str, float]]]:
             # Start dialogue with Agent first.
             for j, utterance in enumerate(dialogue.utterances):
                 if utterance.participant == DialogueParticipant.AGENT.name:
-                    dialogue_utterances_start_agent = [
-                        AnnotatedUtterance.from_utterance(u)
-                        for u in dialogue.utterances[j:]
-                    ]
+                    dialogue_utterances_start_agent = dialogue.utterances[j:]
                     break
```

Comment:
I'm not sure if it is part of this issue but maybe we could refactor this for loop because it is a bit confusing now. We could have something like:
`first_agent_utterance_id = next(utterance_id for utterance_id,utterance in enumerate(dialogue.utterances) if utterance == DialogueParticipant.AGENT.name)`
`dialogue_utterances_start_agent = dialogue.utterances[first_agent_utterance_id:]`
It is just a suggestion.

Reply:
`Utterance` does not have an `utterance_id ` at the moment, we can keep the suggestion for later once the id is available.